packages/react-reconciler/src/ReactFiberHydrationContext.new.js

```diff
@@ -596,6 +616,9 @@ function popHydrationState(fiber: Fiber): boolean {
   if (!supportsHydration) {
     return false;
   }
+  if (enableFloat && isHydratableResource(fiber.type, fiber.memoizedProps)) {
```

Comment:
This is always picking this path if it's a hydratable resource, but the push path is sometimes falling through (if there is no matching resource). This means that they can be misbalanced and lead to something weird.

Reply:
Ah. That's why you check the state node.

Shouldn't it always just insert a node if it's missing to patch things up rather than leaving it wrong?

Reply:
If it is missing, it gets treated like a hydration error and the client side render will patch things up if possible. Sometimes this won't be possible b/c the link will be out of position and I don't yet have insertion semantics for resources (i.e. if you link is a sibling of <html> it will try to insert into the document which will fail)

The least impactful way to do the insertion would be during the hydration itself which feels sorta wrong b/c i'd be mutating the DOM in the render pass. I know we might do this for preloads at some point but that feels a little out of the norm for a hydration insertion.

If I insert via a normal placement I'm going to need to figure out what to do about resources above <html> and it starts to get messy. I'd rather we have a proper Resources implementation that considers hoisting as a first class concept so I was hoping we could leave this alone as a temporary stop gap